google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/SpannerTableStatements.java

```diff
@@ -158,10 +162,22 @@ private boolean isElementCollection(Table table, Metadata metadata) {
 
     statements.add(createTableString);
 
-    // Hibernate requires the special hibernate_sequence table to be populated with an initial val.
     if (table.getName().equals(SequenceStyleGenerator.DEF_SEQUENCE_NAME)) {
-      statements.add("INSERT INTO " + SequenceStyleGenerator.DEF_SEQUENCE_NAME + " ("
-          + SequenceStyleGenerator.DEF_VALUE_COLUMN + ") VALUES(1)");
+      // Caches the INSERT statement in the RunBatchDdl auxiliary object.
+      Optional<RunBatchDdl> runBatchDdl =
+          metadata.getDatabase().getAuxiliaryDatabaseObjects().stream()
+              .filter(obj -> obj instanceof RunBatchDdl)
+              .map(obj -> (RunBatchDdl) obj)
+              .findFirst();
+
+      if (runBatchDdl.isPresent()) {
+        RunBatchDdl runBatchObj = runBatchDdl.get();
+        runBatchObj.addAfterDdlStatement(
+            "INSERT INTO " + SequenceStyleGenerator.DEF_SEQUENCE_NAME + " ("
+                + SequenceStyleGenerator.DEF_VALUE_COLUMN + ") VALUES(1)");
+      } else {
+        LOGGER.warn("Failed to insert the first row into the Hibernate sequence table.");
```

Comment:
Shouldn't you fail hard here because `runbBatchDdl` object is not present? It's not that you were not able to insert the row.

Reply:
Done. It now throws an exception; if this occurs this mean there is some edge case we missed so I added a message to report it as an issue.